packages/react-native-reanimated/src/createAnimatedComponent/AnimatedComponent.tsx

```diff
@@ -374,6 +376,57 @@ export default class AnimatedComponent
     );
   }
 
+  _configureSharedTransition(useNativeId?: boolean) {
+    if (!getStaticFeatureFlag('SHARED_ELEMENT_TRANSITIONS')) {
+      return;
+    }
+    if (this.props.sharedTransitionTag) {
+      const sharedTransition =
+        this.props.sharedTransitionStyle ??
+        this._sharedTransition ??
+        new SharedTransition();
+      if (this._sharedTransition !== sharedTransition) {
```

Comment:
In some cases we'll compare `this._sharedTransition` with `this._sharedTransition` and the condition will be false, is this desired?